src/backend/common/graphics_common.cpp

```diff
@@ -233,34 +233,36 @@ ForgeModule& forgePlugin() { return detail::forgeManager().plugin(); }
 ForgeManager::ForgeManager() : mPlugin(new ForgeModule()) {}
 
 ForgeManager::~ForgeManager() {
-    /* clear all OpenGL resource objects (images, plots, histograms etc) first
-     * and then delete the windows */
-    for (ImgMapIter iter = mImgMap.begin(); iter != mImgMap.end(); iter++)
-        mPlugin->fg_release_image(iter->second);
-
-    for (PltMapIter iter = mPltMap.begin(); iter != mPltMap.end(); iter++)
-        mPlugin->fg_release_plot(iter->second);
-
-    for (HstMapIter iter = mHstMap.begin(); iter != mHstMap.end(); iter++)
-        mPlugin->fg_release_histogram(iter->second);
-
-    for (SfcMapIter iter = mSfcMap.begin(); iter != mSfcMap.end(); iter++)
-        mPlugin->fg_release_surface(iter->second);
-
-    for (VcfMapIter iter = mVcfMap.begin(); iter != mVcfMap.end(); iter++)
-        mPlugin->fg_release_vector_field(iter->second);
-
-    for (ChartMapIter iter = mChartMap.begin(); iter != mChartMap.end();
-         iter++) {
-        for (int i = 0; i < (int)(iter->second).size(); i++) {
-            fg_chart chrt = (iter->second)[i];
-            if (chrt) {
-                mChartAxesOverrideMap.erase((chrt));
-                mPlugin->fg_release_chart(chrt);
+    if (mPlugin->isLoaded()) {
+        /* clear all OpenGL resource objects (images, plots, histograms etc) first
+         * and then delete the windows */
+        for (ImgMapIter iter = mImgMap.begin(); iter != mImgMap.end(); iter++)
+            mPlugin->fg_release_image(iter->second);
```

Comment:
We should have this in a unique_ptr with a custom deleter. This is not the best approach to releasing resources.

Reply:
Agree, but lets do that in different PR. I want to do more changes in forgemanager related code now that I look into it. Please approve the fix to correct segfaults on master. I will send a different PR with my changes to forgemanager that will handle what you suggested also.